src/test/java/brave/opentracing/BraveSpanTest.java

```diff
@@ -145,4 +145,24 @@ public void clear() {
     assertThat(zipkin.spanStore().getDependencies(System.currentTimeMillis(), null))
         .containsExactly(DependencyLink.create("tracer", "tracer2", 1L));
   }
+
+  @Test public void testNotSampled_spanBuilder_parentNull() {
+    tracer.buildSpan("foo")
+            .withTag(Tags.SAMPLING_PRIORITY.getKey(), 0)
+            .start().finish();
+
+    assertThat(zipkin.spanStore().getRawTraces()).isEmpty();
+  }
+
+  @Test public void testNotSampled_spanBuilder_parentNotNull() {
```

Comment:
ps other trace specs are different, like https://github.com/TraceContext/tracecontext-spec.. there may be a time when this sort of on-off pattern in tracing happens in zipkin, just it would need to roll out in a context (no pun intended) that supports it and ideally with multiple tracers agreeing to the rules.